lib/Models/SdmxJsonCatalogItem.js

```diff
@@ -529,39 +575,55 @@ function getTotalValueIdsForDimensionId(item, dimensionId) {
  *             - id
  *             - name
  * If there is a whitelist, only the whitelisted values are included.
+ * If there is a blacklist, blacklisted values are excluded.
  * @private
  * @param  {SdmxJsonCatalogItem} item The SDMX-JSON catalog item.
  * @param  {Array} structureSeries The structure's series property, json.structure.dimensions.series.
  * @return {Object[]} A description of the dimensions.
  */
 function buildDimensions(item, structureSeries) {
-    // Store the length of each dimension, in the correct keyPosition.
-    function getWhitelistFilter(dimensionId) {
-        var thisIdsWhitelist = item.whitelist[dimensionId];
-        if (!defined(thisIdsWhitelist)) {
+    // getFilter returns a function which can be used in list.filter().
+    // It tests if the value is in the given list, using regexps if possible.
+    // filterList can be either item.whitelist or item.blacklist.
+    // set isWhiteList false if is blacklist, so that the return values are negated (except for a missing list).
+    function getFilter(filterList, dimensionId, isWhiteList) {
+        var thisIdsFilterList = filterList[dimensionId];
+        if (!defined(thisIdsFilterList)) {
             return function() { return true; };
         }
         try {
-            var thisIdsWhitelistRegExps = thisIdsWhitelist.map(string => new RegExp('^' + string + '$'));
+            var thisIdsRegExps = thisIdsFilterList.map(string => new RegExp('^' + string + '$'));
```

Comment:
If you're supporting regular expressions, why add the implicit `^` and `$`?  Also, you should mention in the doc for the whitelist and blacklist that it's interpreted as a regular expression.  Otherwise people will be surprised when `.` matches anything rather than requiring a dot.